src/DiskIO/IpcIo/IpcIoFile.cc

```diff
@@ -784,6 +784,9 @@ diskerWriteAttempts(IpcIoMsg &ipcIo)
                    " on " << attempts << " try: " << xstrerr(ipcIo.xerrno));
             ipcIo.len = wroteSoFar;
             return; // bail on error
+        } else {
+            // XXX: setup fd_table[TheFile] properly in DiskerOpen so we can use FD_WRITE_METHOD().
+            fd_table[TheFile].bytesWritten(result);
```

Comment:
Please remove this new `else` and move the bytesWritten() call/comment down, using the wroteNow variable that is computed a few lines below, where the "else" case is already being handled in the official code.

Reply:
done in [3f5541c](https://github.com/squid-cache/squid/pull/1352/commits/3f5541c90c50f6afef45114c34a8915f433d10b2)